fs/bcachefs/chardev.c

```diff
@@ -414,6 +414,11 @@ static long bch2_ioctl_fs_usage(struct bch_fs *c,
 		struct bch_replicas_entry *src_e =
 			cpu_replicas_entry(&c->replicas, i);
 
+		if ((void *)dst_end - (void *)dst_e < sizeof(struct bch_replicas_usage)) {
```

Comment:
I think another solution would be to remove the first `replicas_usage_next` usage and change this to something like

```
if (available_space < sizeof(struct bch_replicas_usage) + sizeof(struct bch_replicas_entry) + 8)
...
```

I think this could work because `nr_devs` will not be set for the following use of `replicas_usage_next`, so we don't actually need to deref the replicas entry

Reply:
I think the existing code is trying to be too clever. Let's just consolidate the two checks into one, open coding replicas_usage_next() so we can do that.

Reply:
:+1: good point it was able to be simplified a good bit